src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -997,11 +997,19 @@ def show_app(cmd, resource_group_name, name, slot=None):
 
 
 def _list_app(cli_ctx, resource_group_name=None):
+    from azure.cli.core.commands.client_factory import get_subscription_id
+    subscription_id = get_subscription_id(cli_ctx)
     client = web_client_factory(cli_ctx)
+    rg_client = resource_groups_factory(cli_ctx, subscription_id)
     if resource_group_name:
         result = list(client.web_apps.list_by_resource_group(resource_group_name))
     else:
-        result = list(client.web_apps.list())
+        resource_groups = rg_client.resource_groups.list()
+        result = []
+        for resource_group in resource_groups:
+            function_apps = list(client.web_apps.list_by_resource_group(resource_group.name))
+            for function_app in function_apps:
+                result.append(function_app)
```

Comment:
Repeated requests for list web_apps based on resource groups may result in poor command performance.

Reply:
> Currently, when the az functionapp list command is run without a resource group, there are some pagination API calls made that fail with a 502 error.

@kaibocai Could we solve the problem with `client.web_apps.list()` on the server side?

Reply:
Thank you for reviewing the PR @zhoxing-ms. We'll get the problem resolved in the server side instead.